app/src/pages/Peers.tsx

```diff
@@ -12,30 +13,46 @@ import {
   GradientHeading,
   PageHead,
   PageWrapper,
-  PeerTable,
+  PeersTable,
 } from '../components';
 
 export const Peers: React.FC = () => {
+  const [isTableLoading, setIsTableLoading] = useState(false);
+
   const { t } = useTranslation();
 
   const dispatch = useAppDispatch();
 
-  useEffect(() => {
-    dispatch(fetchPeers());
-  }, [dispatch]);
-
   const peers = useAppSelector(getPeers);
+  const peersTableOptions = useAppSelector(getPeersTableOptions);
   const peersLoadingStatus = useAppSelector(getPeerLoadingStatus);
 
-  const isLoading = peersLoadingStatus !== Loading.Complete;
+  useEffect(() => {
+    dispatch(fetchPeers(peersTableOptions));
+  }, [dispatch, peersTableOptions]);
+
+  useEffect(() => {
+    if (isTableLoading) {
+      setIsTableLoading(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [peers]);
+
+  const isLoading = peersLoadingStatus !== Loading.Complete && !peers.length;
 
   const pageTitle = `${t('peers')}`;
 
   return (
     <PageWrapper isLoading={isLoading}>
       <PageHead pageTitle={pageTitle} />
       <GradientHeading type="h2">{t('connected-peers')}</GradientHeading>
-      {peers && <PeerTable peers={peers} />}
+      {peers && (
+        <PeersTable
+          peers={peers}
+          isTableLoading={isTableLoading}
+          setIsTableLoading={setIsTableLoading}
+        />
+      )}
```

Comment:
Yeah, let's move the peers fetch into the peers table.  This was it will be easy to refactor in the table skeleton loader instead of the page wrapper loader :)